### PR TITLE
feat: new eth call

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds the new `eth_call` that uses the execution manager `simulateMessage()`
Must be used with the contracts found here: https://github.com/ethereum-optimism/contracts-v2/pull/238

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [ ] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: scripts/start.sh [hunk 1]
```diff
@@ -173,7 +173,7 @@ while (( "$#" )); do
 done
 
 cmd="$REPO/build/bin/geth"
-cmd="$cmd --eth1.syncservice"
+# cmd="$cmd --eth1.syncservice"
```
Comment (tynes) [inline scripts/start.sh]: Uncomment this

File: scripts/start.sh [hunk 2]
```diff
@@ -203,4 +204,4 @@ if [[ ! -z "$IS_VERIFIER" ]]; then
 fi
 
 echo -e "Running:\n$cmd"
-eval env TARGET_GAS_LIMIT=$TARGET_GAS_LIMIT USING_OVM=true $cmd --verbosity=3
+eval env TARGET_GAS_LIMIT=$TARGET_GAS_LIMIT USING_OVM=true $cmd --verbosity=6
```
Comment (tynes) [inline scripts/start.sh]: Change this back to 3

File: core/vm/evm.go [hunk 3]
```diff
@@ -399,6 +394,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 				// empty.
 				success := evm.Context.OriginalTargetResult[:32]
 				ret = evm.Context.OriginalTargetResult[96:]
+				log.Debug("ret", "data", hexutil.Encode(evm.Context.OriginalTargetResult))
```
Comment (tynes) [inline core/vm/evm.go]: Delete this one

File: core/vm/evm.go [hunk 4]
```diff
@@ -363,6 +355,9 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 	}
 
 	ret, err = run(evm, contract, input, false)
+	if err != nil {
```
Comment (tynes) [inline core/vm/evm.go]: Delete this one

File: core/vm/evm.go [hunk 5]
```diff
@@ -270,10 +279,16 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			evm.Context.OriginalTargetAddress == nil {
 			// Whew. Okay, so: we consider ourselves to be at a "target" as long as we were called
 			// by the execution manager, and we're not a precompile or "dead" address.
+			log.Info("target set", "addr", addr.Hex())
```
Comment (tynes) [inline core/vm/evm.go]: Delete this one

File: core/vm/evm.go [hunk 6]
```diff
@@ -97,7 +98,11 @@ func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, err
 			precompiles = PrecompiledContractsIstanbul
 		}
 		if p := precompiles[*contract.CodeAddr]; p != nil {
-			return RunPrecompiledContract(p, input, contract)
+			result, err := RunPrecompiledContract(p, input, contract)
+			if err != nil {
```
Comment (tynes) [inline core/vm/evm.go]: Revert this change

File: core/vm/evm.go [hunk 7]
```diff
@@ -69,7 +70,7 @@ func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, err
 
 		// We don't know the contract, so print some generic information.
 		if isUnknown {
-			log.Debug("Calling Unknown Contract", "Address", contract.Address().Hex())
+			log.Debug("Calling Unknown Contract", "Address", contract.Address().Hex(), "data", hexutil.Encode(input))
```
Comment (tynes) [inline core/vm/evm.go]: Revert this change

**Comment by smartcontracts** [comment]
Scrap lines removed.
